pkg/utils/deployer.go

```diff
@@ -933,3 +933,32 @@ func UpdateDescriptionStatus(desc *appsapi.Description, status *appsapi.Descript
 		return err2
 	})
 }
+
+func BaseUidIndexFunc(obj interface{}) ([]string, error) {
+	base, ok := obj.(*appsapi.Base)
+	if !ok {
+		return nil, fmt.Errorf("object is not a Base %#v", obj)
+	}
+	return []string{string(base.UID)}, nil
+}
+
+func BaseSubUidIndexFunc(obj interface{}) ([]string, error) {
+	base, ok := obj.(*appsapi.Base)
+	if !ok {
+		return nil, fmt.Errorf("object is not a Base %#v", obj)
+	}
+
+	subUid, ok := base.Labels[known.ConfigSubscriptionUIDLabel]
+	if ok {
+		return []string{subUid}, nil
+	}
+	return []string{""}, nil
```

Comment:
This should return non-nil error, instead of nil.

Reply:
done